src/rpp/mutex.cpp

```diff
@@ -75,5 +83,160 @@ namespace rpp
         if (CRITICAL_SECTION* cs = GET_RMUTEX())
             LeaveCriticalSection(cs);
     }
+#elif RPP_FREERTOS
+    /////////////////////////////////////////////////////////////////
+
+    mutex::mutex()
+    {
+        mtx.ctx = xSemaphoreCreateBinary(); // binary semaphore is faster than mutex semaphore
+        xSemaphoreGive((SemaphoreHandle_t)mtx.ctx); // Initialize to available
+    }
+
+    mutex::~mutex()
+    {
+        if (mtx.ctx)
+        {
+            vSemaphoreDelete((SemaphoreHandle_t)mtx.ctx);
```

Comment:
The code is always accessing `mtx.ctx`, so why not just use `void* ctx;` or `void* mtx;`?
The Windows implementation is very specific use case to match WIN32 API SRWLOCK, there is even a static assert to ensure ABI is compatible:
```cpp
static_assert(sizeof(SRWLOCK) == sizeof(mutex::mtx));
```

However here the code would benefit from:
```cpp
#define GET_SEMH() SemaphoreHandle_t(this->ctx)
...
vSemaphoreDelete(GET_SEMH());
```